plotly/colors.py

```diff
@@ -174,21 +176,26 @@ def color_parser(colors, function):
             return new_color_list
 
 
-def validate_colors(colors):
+def validate_colors(colors, colors_list=None):
     """
-    Validates color(s) and returns an error for invalid colors
+    Validates color(s) and returns an error for invalid color(s)
+
+    :param (list) colors_list: whether a singleton color or a list/tuple of
+        colors is inputted, all the color types are appended to colors_list
+        so they can be easily iterated through for validation
     """
-    colors_list = []
+    if colors_list is None:
+        colors_list = []
 
     if isinstance(colors, str):
         if colors in PLOTLY_SCALES:
             return
         elif 'rgb' in colors or '#' in colors:
-            colors_list = [colors]
+            colors_list.append(colors)
```

Comment:
Haha, I should be more clear...

The user inputs `colors` as either a singleton color (i.e. 'rgb(x,y,z)', '#xxyyzz' or (x,y,z)) or an iterable (tuple, list) of these color types. I'm just ensuring here that if a singleton is given, that it's put inside a list so the script can iterate through the list and validate.

I'll add a :param description of `colors` in the String Docs if that helps